src/main/java/jenkins/advancedqueue/priority/strategy/UpstreamCauseStrategy.java

```diff
@@ -64,8 +65,10 @@ private UpstreamCause getUpstreamCause(Queue.Item item) {
 	}
 
 	public int getPriority(Queue.Item item) {
-		int upstreamBuildId = getUpstreamCause(item).getUpstreamBuild();
-		ItemInfo upstreamItem = QueueItemCache.get().getItem(upstreamBuildId);
+		UpstreamCause upstreamCause = getUpstreamCause(item);
+		String upstreamProject = upstreamCause.getUpstreamProject();
+		int upstreamBuildId = upstreamCause.getUpstreamBuild();
+		ItemInfo upstreamItem = StartedJobItemCache.get().getStartedItem(upstreamProject, upstreamBuildId);
```

Comment:
So you do not try to recalculate it if the cache has no item, right? Smells like a bug, because there are use-cases when tasks stay in queue for more than 12 hours